src/visitor.rs

```diff
@@ -817,6 +842,25 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
             snippet_provider: self.snippet_provider,
             macro_rewrite_failure: RefCell::new(false),
             report: self.report.clone(),
+            skip_macro_names: self.skip_macro_names.clone(),
+        }
+    }
+
+    pub fn get_skip_macros(&mut self, attrs: &[ast::Attribute]) {
```

Comment:
I think this function could be simplified by using [`attr`](https://github.com/rust-lang/rust/blob/master/src/libsyntax/attr/mod.rs) module from libsyntax.